src/main/java/frc/robot/subsystems/ConditionedDriveSubsystem.java

```diff
@@ -0,0 +1,39 @@
+package frc.robot.subsystems;
+
+import org.usfirst.frc.team4999.utils.Utils;
+
+public class ConditionedDriveSubsystem extends DriveSubsystem {
```

Comment:
It still doesn't sit right with me that you're using a chain of subsystems for the input processing.
- The docs describe subsystems as "an abstraction for a collection of robot hardware." Using a subsystem for pure data processing that then passes that data to a different subsystem feels like the kind of command misuse we're trying to avoid this year.
- As far as I can tell, calling `require()` with an instance of ConditionedDriveSubsystem doesn't call `require()` with the FalconDriveSubsystem held by ConditionedDriveSubsystem. I guess that wouldn't be too big of an issue if the only reference to the FalconDriveSubsystem was in the ConditionedDriveSubsystem, but the fact that it's passed in as a constructor parameter worries me. I feel there's a great potential that one command could `require()` the ConditionedDriveSubsystem and the other command could `require()` the FalconDriveSubsystem. Since they're two different subsystems the two commands would be allowed to run in parallel, but they would be fighting for the same motors. This could cause unexpected, potentially dangerous behavior.
- The ConditionedDriveSubsystem doesn't use any of the features of SubsystemBase, as far as I can tell. It seems like a waste to give it all the resources and scheduling overhead of a full-blown subsystem, then not use any of it.

Reply:
> @terbos90803 Breaking it out as an independent layer gives you the most flexibility, but also the most overhead. Given the fact that we'll likely need flexibility vs performance, I would recommend going the ConditionedDriveSubsystem route. This allows you to create button mappings and commands that can adjust the speed scaler and then the speed scaler conditions the inputs to the DriveSubsystem.

This is what Terry said before that prompted me to go down this route.  What you see here is the result of a lot of discussion in real life today.  You can ask Terry if you want reasons for pursuing this path.

The decision to make ConditionedDriveSubsystem a Subsystem at all was made because Terry wanted to fulfill the "high cohesion, low coupling" rule by making CDSS and FDSS both impls of an abstract class, DSS.  The advantages of this are well known to you, I'm sure, although I strongly agree that making CDSS a Subsystem is a waste of resources.

To clarify, the current state of the branch is its intended usage.  FalconDriveSS is not intended to be used directly by any Commands that run during teleop, only take input from ConditionedDSS.